tests/commands/record/test_commit.py

```diff
@@ -1,18 +1,47 @@
+from http.server import HTTPServer, SimpleHTTPRequestHandler
+import json
 import os
+import threading
+from typing import Any
 from unittest import mock
 from tests.cli_test_case import CliTestCase
+from launchable.utils.env_keys import BASE_URL_KEY
 from launchable.commands.record.commit import _build_proxy_option
+from tests.helper import ignore_warnings
 
 
-class CommitTest(CliTestCase):
+class CommitHandler(SimpleHTTPRequestHandler):
+    # mock commits/latest
+    def do_GET(self):
+        response = []
+        self.send_response(200)
+        self.send_header("Content-Type", "application/json")
+        self.end_headers()
+        self.wfile.write(json.dumps(response).encode("utf-8"))
+
+    # mock commits/collect
+    def do_POST(self):
+        self.send_response(200)
+        self.send_header("Content-Type", "application/json")
+        self.end_headers()
 
-    @mock.patch.dict(os.environ, {"LAUNCHABLE_TOKEN": CliTestCase.launchable_token})
+
+class CommitTest(CliTestCase):
+    @ mock.patch.dict(os.environ, {"LAUNCHABLE_TOKEN": CliTestCase.launchable_token})
     def test_run_commit(self):
-        """
-        `record commit` command cause error because we can't mock Apache HTTP Client in Java
-        """
-        result = self.cli("record", "commit")
-        self.assertEqual(result.exit_code, 0, "exit normally")
+        server = HTTPServer(("", 0), CommitHandler)
+        thread = threading.Thread(None, server.serve_forever)
+        thread.start()
+
+        host, port = server.server_address
+        endpoint = "http://{}:{}".format(host, port)
```

Comment:
> [](https://sonarcloud.io/project/security_hotspots?id=launchableinc_cli&pullRequest=384&resolved=false&types=SECURITY_HOTSPOT)[](https://sonarcloud.io/project/security_hotspots?id=launchableinc_cli&pullRequest=384&resolved=false&types=SECURITY_HOTSPOT)Security Hotspot E [1 Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=launchableinc_cli&pullRequest=384&resolved=false&types=SECURITY_HOTSPOT)

this warning is here, but this server is used only test so we don't need to fix.